src/http/request.rs

```diff
@@ -61,33 +63,35 @@ impl<'a> Request<'a> {
         Self { body, headers, route }
     }
 
-    pub fn build(&'a self) -> HyperRequestBuilder<'a> {
+    pub fn build(&'a self) -> Result<ReqwestRequestBuilder, HttpError> {
         let Request {
             body,
             headers: ref request_headers,
             route: ref route_info,
         } = *self;
+
         let (method, _, path) = route_info.deconstruct();
 
         let mut builder = CLIENT.request(
-            method.hyper_method(),
-            &path.into_owned(),
+            method.reqwest_method(),
+            Url::parse(&path.to_string())?,
```

Comment:
It should be possible to just do `&path` here, as `Cow` implements `Deref<Target=str>` for `Cow<str>`.
```suggestion
            Url::parse(&path)?,
```